### DISCUSSION TITLE
Purpose/confusion over PrescribeMedication Task

### DISCUSSION THREAD

**Discussion by bevans2000** [mars-sim/mars-sim#1113] [General]
I've been looking at the ``PrecribeMediation``` Task to convert it as part of #1084 but I'm at a loss to see what benefit it provides. It seems to have just 2 condition of stress and radiation poisoning but why are these treated differently and not part of the normal Complaint logic? Then looking further it seems that in ```PhysicalCondition``` it explicitly models 2 conditions, radiation poisoning & stress (although this flag is never changed). Should these just be handled with the existing Complaint logic?
There are only 2 types of medication hard coded but we have a medical.xml that defined treatment plans via a configuration which is the correct way.
Feels to me that this Task does nothing and everything should be controlled by the existing Complaint logic including dealing with stress.
@mokun

**Comment by mokun** [comment]
> it explicitly models 2 conditions, radiation poisoning & stress (although this flag is never changed)

There are a few types of stress induced illnesses, namely, the High Fatigue Collapse, the 

Yes we only need to come up with one single model of Complaint and HealthProblem logic to take care of all illnesses. 

Radiation poisoning should be determined by the dosage of radiation a person receives. As I recall we haven't fully implemented it yet.

But RadiationExposure deals with a person getting a new dose of radiation, it will need a different mechanism to handle.

**EDIT**
See [here](https://github.com/mars-sim/mars-sim/commit/ab60372e7ee672f86dd76e6b7080f927a297a192) for reworking radiation limit and adding more checking if the (30-day, annual, or career) limits has been exceeded, thus triggering radiation poisoning.

**Comment by Urwumpe** [comment]
Shouldn't a prescription rather be a MD instruction to take certain medications? And "take medication" rather be the act of admistering them yourself?

**Reply by mokun** [reply]
You're right. A doctor/pharmacist's job has to do with selecting and approving certain medication be dispensed to a patient.
 
I recall in some Space-opera TV shows or movies, someone would sneak into sickbay to try to steal medications that the dosage won't be approved for him...
